qutebrowser/browser/qutescheme.py

```diff
@@ -165,83 +165,77 @@ def qute_bookmarks(_url):
+        for item in history:
+            # Abort/continue as early as possible
+            item_newer = item.atime > start_time
+            item_older = item.atime < end_time
+            if reverse:
+                # history_dict is reversed, we are going back in history.
+                # so:
+                #     abort if item is older than start_time+24hr
+                #     skip if item is newer than start
+                if item_older:
+                    return
+                if item_newer:
+                    continue
+            else:
+                # history_dict is not reversed, we are going forward in history.
+                # so:
+                #     abort if item is newer than start_time
+                #     skip if item is older than start_time+24hrs
+                if item_older:
+                    continue
+                if item_newer:
+                    return
+
+            # Skip items not within start_time and end_time
             # Skip redirects
             # Skip qute:// links
+            is_in_window = item.atime > end_time and item.atime <= start_time
             is_internal = item.url.scheme() == 'qute'
-            is_not_today = item_atime.date() != curr_date
-            if item.redirect or is_internal or is_not_today:
+            if item.redirect or is_internal or not is_in_window:
                 continue
 
             # Use item's url as title if there's no title.
             item_url = item.url.toDisplayString()
             item_title = item.title if item.title else item_url
-            display_atime = item_atime.strftime("%X")
+            item_time = int(item.atime)
+
+            yield {"url": item_url, "title": item_title, "time": item_time}
 
-            yield (item_url, item_title, display_atime)
+    if QUrl(url).path() == '/data':
+        # Use start_time in query or current time.
+        start_time = QUrlQuery(url).queryItemValue("start_time")
+        start_time = float(start_time) if start_time else time.time()
 
-    if sys.hexversion >= 0x03050000:
-        # On Python >= 3.5 we can reverse the ordereddict in-place and thus
-        # apply an additional performance improvement in history_iter.
-        # On my machine, this gets us down from 550ms to 72us with 500k old
-        # items.
-        history = list(history_iter(reverse=True))
+        if sys.hexversion >= 0x03050000:
+            # On Python >= 3.5 we can reverse the ordereddict in-place and thus
+            # apply an additional performance improvement in history_iter.
+            # On my machine, this gets us down from 550ms to 72us with 500k old
+            # items.
+            history = list(history_iter(start_time, reverse=True))
+        else:
+            # On Python 3.4, we can't do that, so we'd need to copy the entire
+            # history to a list. There, filter first and then reverse it here.
+            history = reversed(list(history_iter(start_time, reverse=False)))
```

Comment:
Use `list(history_iter(start_time, reverse=False))[::-1]` (so it stays a list) for this to work on Python 3.4.